backend/admission.go

```diff
@@ -0,0 +1,114 @@
+package backend
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+)
+
+// AdmissionHandler manages how objects before they are sent to storage
+type AdmissionHandler interface {
+	ValidateAdmission(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+	MutateAdmission(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+	ConvertObject(context.Context, *ConversionRequest) (*AdmissionResponse, error)
```

Comment:
Two things are confusing here:

 - `ValidateAdmission` and `MutateAdmission` have the same signature, which doesn't make sense. For example, the validation hook is allowing to mutate the object, something I would not expect from a validation. Should those requests / responses be different?
 - `MutateAdmission` and `ConvertObject` can also be confused since both allows you to modify the object in the request (and both have the same response struct) so it's not clear if a client should use one or the other.

Reply:
These APIs really follow the k8s conventions -- and can eventually be wired up instead of setting up we webservice for each one 😃 

When configuring the resource callbacks you choose if you want a simple "validate" -- eg, yes no, or "mutate" that will modify the body.  "convert" is called when you need to return the value in a flavor different than what is saved